system/memctrl.c

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (C) 2004-2023 Sam Demeulemeester
+//
+// ------------------------
+//
+// Platform-specific code for IMC configuration, ECC support, etc.
+//
+
+#include <stdbool.h>
+
+#include "config.h"
+#include "cpuinfo.h"
+
+#include "memctrl.h"
+#include "mch/mch.h"
+
+imc_info_t imc;
```

Comment:
I think you need to statically initialise this structure, otherwise you won't get a relocation record for the imc.type field. I needed to do that for the static structures in usbhcd.c.